google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -1227,6 +1226,9 @@ private IOException asIoException(StorageException oops, boolean operationWasCop
 
   @VisibleForTesting
   void doInitStorage() {
-    this.storage = storageOptions.getService();
+    this.storage =
+        storageOptions != null
```

Comment:
why is this being moved to intialization? why can't it be left where it was?

Reply:
@tronto20 for visibility

Reply:
@mpeddada1 thank you

When building a graalvm native-image, `CloudStorageFileSystemProvider` and it's fields are initialized at build time.
(https://github.com/oracle/graal/issues/5134 )
`StorageOptions`, `ServiceOptions`, and their fields are also initialized.

`ServiceOptions` creates credentials, and It determines how to get application default credentials (ADC) in runtime depending on the build time environments.
(see [ServiceOptions](https://github.com/googleapis/sdk-platform-java/blob/cdcc0a201137f3cc93a96cc7670a63d7cc434142/java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L347))
(see [GoogleCredentialsProvider](https://github.com/googleapis/sdk-platform-java/blob/cdcc0a201137f3cc93a96cc7670a63d7cc434142/gax-java/gax/src/main/java/com/google/api/gax/core/GoogleCredentialsProvider.java#L67))

This changes makes..
1. `StorageOptions` and `GoogleCredentialsProvider` to be initialized at the runtime of actual use in graalvm native-image. (not a buildtime)
2. Applications get ADC correctly from runtime environment in graalvm native-image.